### PR TITLE
first step for bd topo unit test

### PR DESCRIPTION
**Pull request by gpetit**
Add a "generic" dataset to test BD Topo v2 case.
The study zone (Gwened) has the `id_zone` = `abcde`

### CODE REVIEW

File: bdtopo_v2/src/test/resources/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/bd_topo_unit_test/bati_indifferencie.prj [hunk 1]
```diff
@@ -0,0 +1 @@
+PROJCS["RGF93_Lambert_93",GEOGCS["GCS_RGF93",DATUM["D_RGF_1993",SPHEROID["GRS_1980",6378137,298.257222101]],PRIMEM["Greenwich",0],UNIT["Degree",0.017453292519943295]],PROJECTION["Lambert_Conformal_Conic"],PARAMETER["standard_parallel_1",49],PARAMETER["standard_parallel_2",44],PARAMETER["latitude_of_origin",46.5],PARAMETER["central_meridian",3],PARAMETER["false_easting",700000],PARAMETER["false_northing",6600000],UNIT["Meter",1]]
```
Comment (ebocher) [inline bdtopo_v2/src/test/resources/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/bd_topo_unit_test/bati_indifferencie.prj:1]: Add a valid epsg prj file

File: bdtopo_v2/src/test/resources/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/bd_topo_unit_test/bati_industriel.prj [hunk 2]
```diff
@@ -0,0 +1 @@
+PROJCS["RGF93_Lambert_93",GEOGCS["GCS_RGF93",DATUM["D_RGF_1993",SPHEROID["GRS_1980",6378137,298.257222101]],PRIMEM["Greenwich",0],UNIT["Degree",0.017453292519943295]],PROJECTION["Lambert_Conformal_Conic"],PARAMETER["standard_parallel_1",49],PARAMETER["standard_parallel_2",44],PARAMETER["latitude_of_origin",46.5],PARAMETER["central_meridian",3],PARAMETER["false_easting",700000],PARAMETER["false_northing",6600000],UNIT["Meter",1]]
```
Comment (ebocher) [inline bdtopo_v2/src/test/resources/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/bd_topo_unit_test/bati_industriel.prj:1]: idem

File: bdtopo_v2/src/test/resources/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/bd_topo_unit_test/bati_remarquable.prj [hunk 3]
```diff
@@ -0,0 +1 @@
+PROJCS["RGF93_Lambert_93",GEOGCS["GCS_RGF93",DATUM["D_RGF_1993",SPHEROID["GRS_1980",6378137,298.257222101]],PRIMEM["Greenwich",0],UNIT["Degree",0.017453292519943295]],PROJECTION["Lambert_Conformal_Conic"],PARAMETER["standard_parallel_1",49],PARAMETER["standard_parallel_2",44],PARAMETER["latitude_of_origin",46.5],PARAMETER["central_meridian",3],PARAMETER["false_easting",700000],PARAMETER["false_northing",6600000],UNIT["Meter",1]]
```
Comment (ebocher) [inline bdtopo_v2/src/test/resources/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/bd_topo_unit_test/bati_remarquable.prj:1]: idem for all

File: bdtopo_v2/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/InputDataFormattingTest.groovy [hunk 4]
```diff
@@ -13,15 +13,14 @@ class InputDataFormattingTest {
 
     def h2GISDatabase
 
-    public static communeToTest = "12174"
+    public static communeToTest = "abcde"
```
Comment (ebocher) [inline bdtopo_v2/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/InputDataFormattingTest.groovy:16]: ?

File: bdtopo_v2/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/InputDataFormattingTest.groovy [hunk 5]
```diff
@@ -134,145 +133,136 @@ class InputDataFormattingTest {
...
 
-        //... with the building 'BATIMENT0000000290116890' : HAUTEUR = 12 / NATURE = 'Chapelle'
+        //... with the building 'BREMAR0003' : HAUTEUR = 12 / NATURE = 'Chapelle'
         assertEquals(12, h2GISDatabase.firstRow("SELECT HEIGHT_WALL FROM BUILDING " +
-                "WHERE ID_SOURCE='BATIMENT0000000290116890';")["HEIGHT_WALL"])
+                "WHERE ID_SOURCE='BREMAR0003';")["HEIGHT_WALL"])
         assertEquals(12, h2GISDatabase.firstRow("SELECT HEIGHT_ROOF FROM BUILDING " +
-                "WHERE ID_SOURCE='BATIMENT0000000290116890';")["HEIGHT_ROOF"])
+                "WHERE ID_SOURCE='BREMAR0003';")["HEIGHT_ROOF"])
         assertEquals(1, h2GISDatabase.firstRow("SELECT NB_LEV FROM BUILDING " +
```
Comment (ebocher) [inline bdtopo_v2/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/InputDataFormattingTest.groovy:163]: You can simplify the test  to return multiple value for the same id

def values = h2GISDatabase.firstRow "SELECT HEIGHT_ROOF, GWEN_GROOVE, JEJE_NOT_WORKING FROM BUILDING    WHERE ID_SOURCE='BATIMENT0000000290116890'"

and do assert on 
values.HEIGHT_ROOF
values.GWEN_GROOVE
values.JEJE_NOT_WORKING

File: bdtopo_v2/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/InputDataFormattingTest.groovy [hunk 6]
```diff
@@ -441,62 +431,62 @@ class InputDataFormattingTest {
...
+        //... with the road 'ROUTE0002' : LARGEUR = 5,5 / NATURE = 'Route à 1 chaussée'
         assertEquals(5.5, h2GISDatabase.firstRow("SELECT WIDTH FROM ROAD " +
-                "WHERE ID_SOURCE='TRONROUT0000000087744143';")["WIDTH"])
+                "WHERE ID_SOURCE='ROUTE0002';")["WIDTH"])
         assertEquals('unclassified', h2GISDatabase.firstRow("SELECT TYPE FROM ROAD " +
-                "WHERE ID_SOURCE='TRONROUT0000000087744143';")["TYPE"])
+                "WHERE ID_SOURCE='ROUTE0002';")["TYPE"])
 
-        //... with the road 'TRONROUT0000000306712994' : LARGEUR = 0 / NATURE = 'Route empierrée'
+        //... with the road 'ROUTE0003' : LARGEUR = 0 / NATURE = 'Route empierrée'
```
Comment (ebocher) [inline bdtopo_v2/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/InputDataFormattingTest.groovy:449]: Idem group all values for the same features

File: bdtopo_v2/src/test/resources/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/bd_topo_unit_test/bati_indifferencie.prj (hunk 1 above)
Comment (gpetit) [reply bdtopo_v2/src/test/resources/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/bd_topo_unit_test/bati_indifferencie.prj:1]: Ok. I will copy / paste the one you pushed for 12174

File: bdtopo_v2/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/InputDataFormattingTest.groovy (hunk 4 above)
Comment (gpetit) [reply bdtopo_v2/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/InputDataFormattingTest.groovy:16]: I just replaced my insee code by this one, to be more "generic" and to remove as many initial informations as possible

File: bdtopo_v2/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/InputDataFormattingTest.groovy (hunk 5 above)
Comment (gpetit) [reply bdtopo_v2/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/InputDataFormattingTest.groovy:163]: Okay, I didn't know that. I'm gonna look into it

Comment (gpetit) [reply bdtopo_v2/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/InputDataFormattingTest.groovy]: Done, with the help from @SPalominos

File: bdtopo_v2/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/InputDataFormattingTest.groovy (hunk 6 above)
Comment (gpetit) [reply bdtopo_v2/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/InputDataFormattingTest.groovy:424]: Done

File: bdtopo_v2/src/test/resources/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/bd_topo_unit_test/bati_indifferencie.prj (hunk 1 above)
Comment (gpetit) [reply bdtopo_v2/src/test/resources/org/orbisgis/orbisprocess/geoclimate/bdtopo_v2/bd_topo_unit_test/bati_indifferencie.prj]: Done

**Comment by ebocher** [comment]
Sounds very good. Is there a way to limit the number of queries. 
For example 

replace

```groovy
        //... with the rail 'RAIL0002' : NATURE = 'Principale'
        assertEquals('rail', h2GISDatabase.firstRow("SELECT TYPE FROM RAIL " +
                "WHERE ID_SOURCE='RAIL0002';")["TYPE"])

        // Check if rails are well selected or not ...
        // ... with the rail 'RAIL0002' which is inside the zone --> so expected 1
        assertEquals(1, h2GISDatabase.firstRow("SELECT COUNT(*) as TOTAL FROM RAIL " +
                "WHERE ID_SOURCE='RAIL0002';")["TOTAL"])
```

by

```groovy
def data = h2GISDatabase.firstRow("SELECT TYPE,  COUNT(*) as TOTAL FROM RAIL, TYPE WHERE ID_SOURCE='RAIL0002';"
assertEquals('rail', data.TYPE);
assertEquals(1, data.TOTAL);
```

It will be easier also to write the tests. You can create a generic method like this

def getData(def id_source){
return  h2GISDatabase.firstRow("SELECT TYPE,  COUNT(*) as TOTAL FROM RAIL, TYPE WHERE ID_SOURCE=?;", id_source)
}
then

def data =  getData("RAIL0002");
assertEquals('rail', data.TYPE);
assertEquals(1, data.TOTAL);

def data =  getData("RAIL0001");
assertEquals('gwen_on_the_road', data.TYPE);
assertEquals(1, data.TOTAL);

**Comment by gpetit** [comment]
Hum ... it will be easier for you ... but not for me ;-)

**Comment by ebocher** [comment]
It will be easier to read the code and understand the tests.

**Pull request merged**